### PR TITLE
Fix ViewPager crash

### LINKED ISSUE #3076: [BUG] Google Play crash (related with dispatchTouchEvent)
Reported from Google Play:

```
java.lang.IllegalArgumentException: 
  at android.view.MotionEvent.nativeGetAxisValue (Native Method)
  at android.view.MotionEvent.getX (MotionEvent.java:2762)
  at androidx.viewpager.widget.ViewPager.onInterceptTouchEvent (ViewPager.java:2072)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:3400)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:3863)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:3551)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:3863)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:3551)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:3863)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:3551)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:3863)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:3551)
  at android.view.ViewGroup.dispatchTransformedTouchEvent (ViewGroup.java:3863)
  at android.view.ViewGroup.dispatchTouchEvent (ViewGroup.java:3551)
  at com.android.internal.policy.DecorView.superDispatchTouchEvent (DecorView.java:770)
  at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent (PhoneWindow.java:1952)
  at android.app.Activity.dispatchTouchEvent (Activity.java:4038)
  at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent (WindowCallbackWrapper.java:69)
  at com.android.internal.policy.DecorView.dispatchTouchEvent (DecorView.java:728)
  at android.view.View.dispatchPointerEvent (View.java:14636)
  at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent (ViewRootImpl.java:6582)
  at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess (ViewRootImpl.java:6360)
  at android.view.ViewRootImpl$InputStage.deliver (ViewRootImpl.java:5796)
  at android.view.ViewRootImpl$InputStage.onDeliverToNext (ViewRootImpl.java:5849)
  at android.view.ViewRootImpl$InputStage.forward (ViewRootImpl.java:5815)
  at android.view.ViewRootImpl$AsyncInputStage.forward (ViewRootImpl.java:5971)
  at android.view.ViewRootImpl$InputStage.apply (ViewRootImpl.java:5823)
  at android.view.ViewRootImpl$AsyncInputStage.apply (ViewRootImpl.java:6028)
  at android.view.ViewRootImpl$InputStage.deliver (ViewRootImpl.java:5796)
  at android.view.ViewRootImpl$InputStage.onDeliverToNext (ViewRootImpl.java:5849)
  at android.view.ViewRootImpl$InputStage.forward (ViewRootImpl.java:5815)
  at android.view.ViewRootImpl$InputStage.apply (ViewRootImpl.java:5823)
  at android.view.ViewRootImpl$InputStage.deliver (ViewRootImpl.java:5796)
  at android.view.ViewRootImpl.deliverInputEvent (ViewRootImpl.java:9032)
  at android.view.ViewRootImpl.doProcessInputEvents (ViewRootImpl.java:8889)
  at android.view.ViewRootImpl.enqueueInputEvent (ViewRootImpl.java:8842)
  at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent (ViewRootImpl.java:9169)
  at android.view.InputEventReceiver.dispatchInputEvent (InputEventReceiver.java:194)
  at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents (Native Method)
  at android.view.InputEventReceiver.consumeBatchedInputEvents (InputEventReceiver.java:183)
  at android.view.ViewRootImpl.doConsumeBatchedInput (ViewRootImpl.java:9109)
  at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run (ViewRootImpl.java:9196)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:999)
  at android.view.Choreographer.doCallbacks (Choreographer.java:797)
  at android.view.Choreographer.doFrame (Choreographer.java:725)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:984)
  at android.os.Handler.handleCallback (Handler.java:883)
  at android.os.Handler.dispatchMessage (Handler.java:100)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8167)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
```

it happens in different devices and Android versions.

### PR DESCRIPTION
**Pull request by abelgardep**
Fixes #3076

### CODE REVIEW

**Comment by davigonz** [comment]
Since 2013, they could have fixed it in any of the last versions 😄

**Review by davigonz — APPROVED** [review]

**Comment by jesmrec** [comment]
Checked with [these steps](https://github.com/owncloud/android/issues/3076#issuecomment-779723597), and not reproducible anymore. 

Let's try in the next release if the crash is out of GP.

File: owncloudApp/src/main/java/com/owncloud/android/ui/preview/ViewPagerWorkAround.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class ViewPagerWorkAround : ViewPager {
+    constructor(context: Context) : super(context)
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {}
+
+    @SuppressLint("ClickableViewAccessibility")
+    override fun onTouchEvent(ev: MotionEvent?): Boolean {
+        try {
+            return super.onTouchEvent(ev)
+        } catch (ex: IllegalArgumentException) {
+            ex.printStackTrace()
```
Comment (hannesa2) [inline owncloudApp/src/main/java/com/owncloud/android/ui/preview/ViewPagerWorkAround.kt:21]: Consider to use `Timber.e(ex)` when you want to see it in logfil. This is more a `/dev/null`

Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/ui/preview/ViewPagerWorkAround.kt:21]: Yes, you are right!

**Pull request merged**